### ISSUE TITLE
Auto MDM Enrollment - Differences between User/Device Credentials

### ISSUE THREAD

**Issue by v-maxist** [MicrosoftDocs/windows-itpro-docs#4828]
Hi there!

On Windows 1709, there is the option of using "Auto MDM Enrollment with AAD Token" (As currently documented).
However, starting with Windows 1903, the GPO is now called "Enable automatic MDM enrollment using default Azure AD credentials", and we have the option to choose either User/Device Credentials.

Could the doc be updated to clarify the differences between User and Device Credentials?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ad012591-5c74-f1cd-d3a2-443768e8ceee
* Version Independent ID: 918ea4bc-0849-3745-5d9d-a21a973bcd3b
* Content: [Enroll a Windows 10 device automatically using Group Policy](https://docs.microsoft.com/en-us/windows/client-management/mdm/enroll-a-windows-10-device-automatically-using-group-policy#)
* Content Source: [windows/client-management/mdm/enroll-a-windows-10-device-automatically-using-group-policy.md](https://github.com/MicrosoftDocs/windows-itpro-docs/blob/master/windows/client-management/mdm/enroll-a-windows-10-device-automatically-using-group-policy.md)
* Product: **w10**
* Technology: **windows**
* GitHub Login: @ManikaDhiman
* Microsoft Alias: **dansimp**

**Comment by jvsam** [comment]
@officedocsbot assign @jvsam

**Comment by jvsam** [comment]
assign @jvsam

**Comment by jvsam** [comment]
Hello @v-maxist thank you for bringing this to our attention. Your feedback is appreciated. 

Hello @ManikaDhiman, one of the requirements stated on this doc is *AD-joined PC running Windows 10, version 1709 or later*, however, as stated by Max, on 1903, the option under MDM is different using GP. I can confirm it on my end, since this is what I see. 

![image](https://user-images.githubusercontent.com/18371314/64255951-fe2bda00-cf54-11e9-9ab8-6c2ea2717003.png)

Can you please confirm if this is the case for 1903 and later? Should we add a note to address this? Thank you.

**Comment by jvsam** [comment]
> ...one of the requirements stated on this doc is AD-joined PC running Windows 10, version 1709 or later, however, as stated by Max, on 1903, the option under MDM is different using GP. I can confirm it on my end, since this is what I see.

Hi again @ManikaDhiman, do you have any update on this? I believe it will be best if we add a note—if confirmed—regarding the different steps or options for 1903. What do you think? Thank you.

**Comment by jvsam** [comment]
Hello again @ManikaDhiman just following up if you have information or update regarding this. Thank you.

**Comment by mrik23** [comment]
Yes the difference between user and device credentials needs to be explained. And how to troubleshoot issue with device credential. For example I'm getting the error below, but it doesn't make much sense.

"MDM Enroll: Server Returned Fault/Code/Subcode/Value=(MessageFormat) Fault/Reason/Text=(Device based token is not supported for enrollment type OnPremiseGroupPolicyCoManaged)."

**Comment by byteben** [comment]
Agree that this needs more explanation. The assumption is the behavior will be similar to when using co-management where the device will enroll into Intune and get a Device Owner as soon as an Intune licenced user logs on. However, in our testing, the device does not enroll into Intune with the device token "unless" an Intune licenced user is logged on - this kind of goes against my interpretation of the GPO. My understanding is the device should enroll into Intune with the AAD Device token "before" a licenced Intune user logs in, similar to what happens in co-management in SCCM 1906.

Certainly more clarification and documentation is required.

**Comment by ManikaDhiman** [comment]
Hi @v-maxist, I am awaiting response from the product team on this issue. I will update the doc as soon as I hear from them. Thanks!

**Comment by jvsam** [comment]
Thank you for the update @ManikaDhiman. Looking forward to the feedback of the product team.

**Comment by lXbalanque** [comment]
Hi all, I'm throwing my hat in the ring here as well. I've been trying to get Hybrid AAD Joins working in conjunction with this feature, to automatically enroll my devices but I don't know what to select. What is the difference between device and user credential? Looking forward to a solution/detailed reply. The current document does appear to be outdated as it names a policy that does not exist or, at the very least, is not worded the same.

**Comment by byteben** [comment]
Hey,

MDM enrolment always used to run in the user context. This would happen when an Intune licenced user logs on to a Windows 10 PC.

The GPO was updated for 1903 to give the additional option of using the device token to enroll the PC. In theory, the PC would enroll into Intune BEFORE the user logs in, speeding up enrolment of the device and subsequent deployment of policies. This is similar to the behaviour we observe with SCCM 1906 co-management.

The device enrolment with the AAD device token was working up until a few weeks ago. This thread was started to try and get better docs to help us understand the mechanisms of Device token enrolment and to troubleshoot it when it goes wrong.

Unlike others, my devices do enroll with the AAD Device token but ONLY after an Intune licenced user logs on..which is just bizarre.

The docs team are working on this, more info is coming, but a lot of us are screaming for it sooner rather than later.

Hope that helps.

If you have SCCM I would encourage you to look at co-management. If not, choose the option to enroll as the user for now.

Kind Regards
Ben

Sent from my iPhone

On 26 Sep 2019, at 20:30, lXbalanque <notifications@github.com> wrote:

﻿

Hi all, I'm throwing my hat in the ring here as well. I've been trying to get Hybrid AAD Joins working in conjunction with this feature, to automatically enroll my devices but I don't know what to select. What is the difference between device and user credential? Looking forward to a solution/detailed reply. The current document does appear to be outdated as it names a policy that does not exist or, at the very least, is not worded the same.

**Comment by lXbalanque** [comment]
Hi Ben,

Thanks for your reply. My issue is specifically with Intune MDM enrollment. I don't use SCCM at all but I appreciate the additional detail you've brought to the discussion. Device enrollment would be ideal and the way I think most would prefer as it speeds up the entire provisioning process. You're not stuck waiting for a user to log into the machine and enrollment occurs sooner rather than later. I have a test scenario right now where I finally got Hybrid AADJ working but need to enable this policy. 

I think at the moment, I'm going to at least try the device credential as its the configuration I'd like to eventually end up with. If it doesn't work I'll test the other way around.

Looking forward to an update from the doc team.

> Hey, MDM enrolment always used to run in the user context. This would happen when an Intune licenced user logs on to a Windows 10 PC. The GPO was updated for 1903 to give the additional option of using the device token to enroll the PC. In theory, the PC would enroll into Intune BEFORE the user logs in, speeding up enrolment of the device and subsequent deployment of policies. This is similar to the behaviour we observe with SCCM 1906 co-management. The device enrolment with the AAD device token was working up until a few weeks ago. This thread was started to try and get better docs to help us understand the mechanisms of Device token enrolment and to troubleshoot it when it goes wrong. Unlike others, my devices do enroll with the AAD Device token but ONLY after an Intune licenced user logs on..which is just bizarre. The docs team are working on this, more info is coming, but a lot of us are screaming for it sooner rather than later. Hope that helps. If you have SCCM I would encourage you to look at co-management. If not, choose the option to enroll as the user for now. Kind Regards Ben
> […](#)

**Comment by tlfzhylj** [comment]
The option to choose between _Device credential_ and _User credential_ is also present in Windows 10 1809.

**Comment by mrik23** [comment]
We still need clarification on this. And device credential stopped working with 1903 for few weeks now.

**Comment by jvsam** [comment]
Thanks for the additional information @tlfzhylj. FYI @ManikaDhiman.

> And device credential stopped working with 1903 for few weeks now...

Yes @mrik23, the doc author is already coordinating with the Product team to get more details on *Device credential* and *User credential*, and ultimately, to update this documentation.  As per this doc, *Device credential* should work on Windows 10, version 1903, can you also please check @ManikaDhiman?

Thank you.

**Comment by jvsam** [comment]
assign @ManikaDhiman

**Comment by ManikaDhiman** [comment]
Hi @jvsam, Updates for User/Device Credentials were added via PR#4969, which are now merged with live docs. Hopefully, those updates will resolve this issue. Thanks!

**Comment by jvsam** [comment]
Hi @ManikaDhiman, thank you for the update.

@v-maxist. @mrik23, @byteben, @lXbalanque good news, the doc has been updated. It now reflects the correct screens and instructions.  Please see PR #4969 for the specific updates. 

@tlfzhylj please check the [Configure the auto-enrollment for a group of devices](https://docs.microsoft.com/windows/client-management/mdm/enroll-a-windows-10-device-automatically-using-group-policy#configure-the-auto-enrollment-for-a-group-of-devices) section. The important note provides information in fixing the missing policy issue. 

Again, thank you everyone for your feedback. We will now close this issue. Please continue to send your feedback and suggestions to improve the quality of the Microsoft documentation.

**Comment by jvsam** [comment]
@officedocsbot close

**Comment by illfated** [comment]
https://github.com/MicrosoftDocs/windows-itpro-docs/issues/4828#issuecomment-547944123

"Good news, everyone!" 😄

![image](https://user-images.githubusercontent.com/38162891/67897035-df655080-fb5d-11e9-91b3-51d3cce372c2.png)

**Comment by megaalc** [comment]
hi all, is that supported in WVD 1909 and/or 2004 win 10 ? getting same error in my multisession WVD 1909 environment. DOCs not specifically mentioning 1909 / 2004 support.

**Comment by illfated** [comment]
@megaalc : You may want to open a new issue ticket for that, unless you find that this old and closed ticket contains any particularly helpful comments.

Please also note that the MS Docs team member ticket handler (@jvsam ) is no longer working here in this repository or with the MS Docs team.

Disclaimer: I am not affiliated with Microsoft, not here on GitHub or elsewhere. I am simply trying to help others, if and when I can.

**Comment by megaalc** [comment]
nah, do not want to open anything new, just thought may be it is worth asking here whether same doc is already outdated or not..
all good

**Comment by gerrynicol** [comment]
> nah, do not want to open anything new, just thought may be it is worth asking here whether same doc is already outdated or not..
> all good

megaalc Did you download the admx files for W10 2004 to get it working?

**Comment by megaalc** [comment]
Unfortunatelly only using "User Credentials" option (not device credentials) which is not what i wanted  , and also i gave up playing with it  as  Intune is not officially supported in win10 WVD environamnt

**Comment by gerrynicol** [comment]
> Unfortunatelly only using "User Credentials" option (not device credentials) which is not what i wanted , and also i gave up playing with it as Intune is not officially supported in win10 WVD environamnt

Thanks for replying. Fair enough. I'll do some more testing

**Comment by drenergy** [comment]
Hi,

Is there any update regarding the device credential option? I cannot find if this is already being supported or should at least work. I have tested this multiple times, but it always falls back to user credentials as soon as log on.

I expected this to work without user interaction.

**Comment by illfated** [comment]
Wish I could help, but this repository is not intended for regular support issues, support requests or support questions.

You may want to look elsewhere for answers, unless you have suggestions for improving one or more specific pages in the MS Docs repository.

Disclaimer: I am not affiliated with Microsoft or the Microsoft Docs teams here on GitHub.

The intention of this repository is to improve on the existing documentation by providing the users a way to report factual errors, improvement suggestions and request updates for new versions.

The MS Docs team members might not have time to answer your question, both due to the small size of the team and the big workload.

Try the following resources to get in touch with users who may be able to help you:
- TechNet forums: https://social.technet.microsoft.com/Forums/en-US/home
- MSDN: https://social.msdn.microsoft.com/Forums/en-US/home
- Microsoft Community: https://answers.microsoft.com/

**Comment by MaxSteinMS** [comment]
Hi @drenergy, thanks for taking the time to share your experience.
Device Credential is not supported for GPO enrollment into Intune, and only User Credential is currently supported.

In addition to the resources illfated mentioned, if you are having any further issues with the Intune side of things, there are a few options available to provide the fastest level of support:

- Use [How to get support for Microsoft Intune](http://aka.ms/IntuneSupport) to open a case with professional support.
- Post on [Microsoft Intune forums](https://techcommunity.microsoft.com/t5/microsoft-intune/bd-p/Microsoft-Intune). These are moderated by our community MVPs who are very experienced, knowledgeable, and helpful.
- Contact your Microsoft Premier team, such as a Premier Field Engineer or Technical Account Manager.

**Comment by illfated** [comment]
Thank you for the nice reply. Well said.
